tests/unit/util/test_deprecate_args.py

```diff
@@ -0,0 +1,205 @@
+"""This file should be updated as files/classes/functions are deprecated."""
+import pytest
+
+from praw.util import _deprecate_args
+
+from .. import UnitTest
+
+
+@_deprecate_args("arg2", "arg1", "arg3", "arg0")
+def arg_test_global(*, arg0=None, arg1=None, arg2=None, arg3=None):
+    return arg0, arg1, arg2, arg3
+
+
+@_deprecate_args("arg0", "arg2", "arg1", "arg3")
+def arg_test_global_with_positional(arg0, *, arg1=None, arg2=None, arg3=None):
+    return arg0, arg1, arg2, arg3
+
+
+class ArgTest:
+    @_deprecate_args("arg2", "arg1", "arg3", "arg0")
+    def arg_test(self, *, arg0=None, arg1=None, arg2=None, arg3=None):
+        return arg0, arg1, arg2, arg3
+
+    @_deprecate_args("arg0", "arg2", "arg1", "arg3")
+    def arg_test_with_positional(self, arg0, *, arg1=None, arg2=None, arg3=None):
+        return arg0, arg1, arg2, arg3
+
+
+class TestDeprecateArgs(UnitTest):
+    def setup(self):
+        self.arg_test = ArgTest()
+
+    def test_arg_test(self):
+        with pytest.warns(
+            DeprecationWarning,
+            match=(
+                "Positional arguments for 'ArgTest.arg_test' will no longer be"
+                " supported in PRAW 8.\nCall this function with 'arg2', 'arg1', 'arg3',"
+                " and 'arg0' as keyword arguments."
+            ),
+        ):
+            result_args = self.arg_test.arg_test("arg2", "arg1", "arg3", "arg0")
+            assert result_args == ("arg0", "arg1", "arg2", "arg3")
+
+    def test_arg_test__keyword(self):
+        result_args = self.arg_test.arg_test(
```

Comment:
Should this assert no warning occurs, or would the test fail if an uncaught warning occurs?

Reply:
Good point. It should probably test that no warning is raised because I don't believe it will fail if it does. This test is the make sure that all the keyword args are mapped properly.